backend/stakepoold/rpcclient.go

```diff
@@ -110,70 +112,170 @@ func connectWalletRPC(cfg *config) (*dcrrpcclient.Client, semver, error) {
 	return dcrwClient, walletVer, nil
 }
 
-func walletFetchUserTickets(ctx *appContext) (map[chainhash.Hash]string, string) {
-	// This is suboptimal to copy and needs fixing.
-	users := make(map[string]userdata.UserVotingConfig)
-	ctx.RLock()
-	for k, v := range ctx.userVotingConfig {
-		users[k] = v
+func nodeCheckTicketFee(ctx *appContext, txhex string, blockhash string) (bool, error) {
+	// Create raw transaction.
+	var buf []byte
+	buf, err := hex.DecodeString(txhex)
+	if err != nil {
+		return false, fmt.Errorf("DecodeString failed: %v", err)
+	}
+	res, err := ctx.nodeConnection.DecodeRawTransaction(buf)
+	if err != nil {
+		return false, fmt.Errorf("DecodeRawTransaction failed: %v", err)
 	}
-	ctx.RUnlock()
 
-	userTickets := make(map[chainhash.Hash]string)
+	bhash, err := chainhash.NewHashFromStr(blockhash)
+	if err != nil {
+		return false, fmt.Errorf("NewHashFromStr failed for %v: %v", blockhash, err)
+	}
+	gbh, err := ctx.nodeConnection.GetBlockHeader(bhash)
+	if err != nil {
+		return false, fmt.Errorf("GetBlockHeader failed for %v: %v", bhash, err)
+	}
 
-	type promise struct {
-		dcrrpcclient.FutureStakePoolUserInfoResult
-		msa string
+	// Check the first commitment output (Vout[1])
+	// and ensure that the address found there exists
+	// in the list of approved addresses. Also ensure
+	// that the fee exists and is of the amount
+	// requested by the pool.
+
+	fees := dcrutil.Amount(0)
+	feeAddr := ""
+	feeAddrValid := false
+	ticketValue, err := dcrutil.NewAmount(res.Vout[0].Value)
+	if err != nil {
+		return false, fmt.Errorf("NewAmount failed: %v", err)
+	}
+
+	for i := range res.Vout {
+		switch res.Vout[i].ScriptPubKey.Type {
+		case "sstxcommitment":
+			for j := range res.Vout[i].ScriptPubKey.Addresses {
+				_, exists := ctx.feeAddrs[res.Vout[i].ScriptPubKey.Addresses[j]]
+				if exists {
+					feeAddr = res.Vout[i].ScriptPubKey.Addresses[j]
+					feeAddrValid = true
+					feeAmount, err := dcrutil.NewAmount(*res.Vout[i].ScriptPubKey.CommitAmt)
+					if err != nil {
+						return false, fmt.Errorf("NewAmount failed: %v", err)
+					}
+					fees = feeAmount
+				}
+			}
+		}
+	}
+
+	in := dcrutil.Amount(0)
+	for i := range res.Vout {
+		switch res.Vout[i].ScriptPubKey.Type {
+		case "sstxcommitment":
+			amount, err := dcrutil.NewAmount(*res.Vout[i].ScriptPubKey.CommitAmt)
+			if err != nil {
+				return false, fmt.Errorf("NewAmount failed: %v", err)
+			}
+			in += amount
+		}
 	}
-	promises := make([]promise, 0, len(users))
-	for msa, v := range users {
-		addr, err := dcrutil.DecodeAddress(msa, ctx.params)
+
+	out := dcrutil.Amount(0)
+	for i := range res.Vout {
+		amount, err := dcrutil.NewAmount(res.Vout[i].Value)
 		if err != nil {
-			log.Infof("Could not decode multisig address %v for %v: %v",
-				msa, v.Userid, err)
-			continue
+			return false, fmt.Errorf("NewAmount failed: %v", err)
 		}
+		out += amount
+	}
+	fees2 := in - out
```

Comment:
I updated the comment to reflect that this function is ported from wallet where it does have some tests (but more for the underlying functions/calculations).  Needs some more though.